client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx

```diff
@@ -29,39 +32,41 @@ interface ProfilePhotoProps {
 const ProfilePhoto: React.FC<ProfilePhotoProps> = ({ header, currentUser, currentProfile }) => {
   const classes = useStyles();
   const { updateSnackBarMessage } = useSnackBar();
-
-  const [file, setFile] = useState();
-  const [images, setImages] = useState([]);
   const [isSubmitting, setSubmitting] = useState(false);
+  const [imageKey, setImagesKey] = useState(currentProfile.photoKey);
 
   const fileSelected = (event: React.ChangeEvent<HTMLInputElement>) => {
     const files = event.target.files!;
     setSubmitting(true);
     const promises = [];
     for (let i = 0; i < files.length; i++) {
       promises.push(postPhoto(files[i]));
-      Promise.all(promises)
-        .then((dataArray) => {
-          for (let j = 0; j < dataArray.length; j++) {
-            const data = dataArray[j];
-            if (data.error) {
-              console.error({ error: data.error.message });
-              updateSnackBarMessage(data.error.message);
-            } else if (data.success) {
-              const values = { photo: data.success.imagePath };
-              editProfilePhotoUri(values);
-              updateSnackBarMessage('Photo updated!');
-            } else {
-              // should not get here from backend but this catch is for an unknown issue
-              console.error({ data });
-              updateSnackBarMessage('An unexpected error occurred. Please try again');
-            }
-          }
-        })
-        .then(() => {
-          setSubmitting(false);
-        });
     }
+    Promise.all(promises)
+      .then((dataArray) => {
+        for (let j = 0; j < dataArray.length; j++) {
```

Comment:
Since we would only select one profile photo each time, I don't think we'll need to support multiple photo uploads through this page :)